### PR TITLE
[UPDATE] Table List Item Customization, Form Delete show

### PR DESCRIPTION
**Pull request by FrankyMartz**
## Description

* Create table attribute `canEditItem` for per table item edit control
* Create table attribute `canDeleteItem` for per table item delete control
* Create table attribute `showEditDelete` hiding dialog edit delete button
* Create form attribute `showDelete` hiding dialog delete button
* Attach Table List Item index to each table row for targeted styling
* Update (README) documentation

## Motivation and Context
__Why is this change required? What problem does it solve?__
Previous Roadhouse Plugin state was limited in terms of customization. This update is meant to expanded customization capability.

## How Has This Been Tested?

* Basic server API requests verified in local development environment
* Run existing specs against local development environment

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] Breaking change (fix or feature that would cause existing functionality to change)
- - -

@vokal/web

### CODE REVIEW

**Comment by FrankyMartz** [comment]
Travis-CI configuration requires attention. It appears as though the tests are failing due to Selenium Web driver.

**Comment by FrankyMartz** [comment]
Updates made pending Review and Merge
@vokal/web

File: source/directives/rhTable.js [hunk 1]
```diff
@@ -34,6 +38,9 @@ function ( $compile, $rootScope, ngDialog, alertify )
                 scope.loading = true;
             }
 
+            scope.canEditItem = angular.isFunction(scope.canEditItem) ? scope.canEditItem : function() { return true };
+            scope.canDeleteItem = angular.isFunction(scope.canDeleteItem) ? scope.canDeleteItem : function() { return true };
```
Comment (jrit) [inline source/directives/rhTable.js]: missing `;` within function bodies after 'true`

File: source/directives/rhForm.js [hunk 2]
```diff
@@ -43,6 +44,17 @@ module.exports = [ "$compile", function ( $compile )
                         scope.save( scope.model );
                     }
                 };
+                scope.getShowDeleteState = function ()
+                {
+                    // NOTE: Maintain Default for Previous Implementation
+                    if ( !(typeof( scope.showDelete ) === 'boolean' || angular.isFunction( scope.showDelete )) )
+                    {
+                        return true;
+                    }
```
Comment (jrit) [inline source/directives/rhForm.js]: I'm not sure even with the comment that I understand what this block is for. If it makes sense to have a breaking change, that's fine, we'll just bump the major version for release

Comment (FrankyMartz) [inline source/directives/rhForm.js]: It was my attempt at limiting the `rhForm` directive argument `showDelete` to either a (`Function` or `Boolean`), and returning `true` as a default fallback

**Comment by FrankyMartz** [comment]
Ready for re-review. Updates are non-destructive and should not affect existing implementation.

**Review by jrit — APPROVED** [review]

**Pull request merged**